### PR TITLE
fix for dsa key size feature request issue: pkey -text or -text_pub s…

### PR DESCRIPTION
**Pull request by jayaram-intel**
Fixes for  #9927  to address dsa key size feature request issue.

**Output of `openssl pkey -text_pub -pubout` command:**

$ openssl dsaparam -out dsaparam.pem 2048
$ openssl gendsa -out privkey.pem dsaparam.pem
$ **`openssl pkey -text_pub -pubout  < privkey.pem`**
-----BEGIN PUBLIC KEY-----
MIIDRzCCAjoGByqGSM44BAEwggItAoIBAQDDH5FNEsYfotNms1VyZ30qmi2sH+dl
N2Tu9ug3TQvGWZjVpxqUwdsel1s7FNkyWNrS85oMpIhQWPAS0rQNNvWlGJ9i7D9q
W5BMACfyUx2EumrRjd7Xgwl+fP3H3uR14C+Cy+a2zTPBp4Xj8Ar7W5vpz/JsBl77
uTtuFXYA8U7O4zpF/B1HmBAEtqS8yioc2IIX+ke9z+Gcru1dJSIKQ7g64f/RKwnv
uPhlAyLrxsT8CNc3bOq0jqDJCl7RcITVOFLxWX+dAaGFQ0TmMWgAs/EbPC2FEXnd
NyfbT+3XYg3vVCQUv/nG/sBauFu/p3ElAUz5ICPdiAxRryeEzMjt3NRDAiEAiNQ7
moZSi8xpJBt78eZV1+ARlns1XTKu7X0FiZo84rMCggEBAKuPtZO8DRAVvJ61Lovz
Z+9oYV07ynhETZFkUC/IqO3GYFD7ivEiPlqzsh1stKp4DsBLldoRxDZafc2+JFA5
k0DuBn4Lg7a3/oCUhioUCAgRja2iCUrzmgHgjbg+tk/S2VO84x7RfXU7LMquILge
77fcGBrQXTBG3N+aC89crXdB/x2pxqcjrsx6frUDnVEb/cAM+w7ttgPykq/rHnDF
I7Os3Mz9hrQxGYRfI08W9KzLC7p3RsE6D7FpNqtBRadBjUTbzwfca4D9yovm/de5
AEUDhvmFsbqwR8OOo0BCfMg4lTCo+1Do7pjjKwM5nbcHoheGVc3vowJWpTDPDI4V
czUDggEFAAKCAQBsFJKjnsrqs+GkXjZfxfFEU414T84F4MDNHe7HO/YIP6PQsQhr
/OKMGtyDARxXAnlpVv29NQZO4sI3ylZPtbJitEifmXf+I/8O6laVI1ixwEyfBnTu
eaSx48rrocPQB9d8N2JGAQKKmMmJ5dSjLlu7Yf8zjm/0/nyIs5zi/wMTC09zoTLy
LuJMnXnLKDgTgfPYEtd7gBJaAYal2fUbOpkvKGfKjBAjBGLrjqwFPEVLk1GYVqg5
LUHswAIkyB3kfOy7y2t5aZERf14Y46vkriG4a+G0RlJfFpdes2tit97Ofj3isSgc
cm3NJwoiZlXPd3XYdOLq4o2v0pncHkttu8S9
-----END PUBLIC KEY-----
Public-Key: (2048 bit)
pub: 
    6c:14:92:a3:9e:ca:ea:b3:e1:a4:5e:36:5f:c5:f1:
    44:53:8d:78:4f:ce:05:e0:c0:cd:1d:ee:c7:3b:f6:
    08:3f:a3:d0:b1:08:6b:fc:e2:8c:1a:dc:83:01:1c:
    57:02:79:69:56:fd:bd:35:06:4e:e2:c2:37:ca:56:
    4f:b5:b2:62:b4:48:9f:99:77:fe:23:ff:0e:ea:56:
    95:23:58:b1:c0:4c:9f:06:74:ee:79:a4:b1:e3:ca:
    eb:a1:c3:d0:07:d7:7c:37:62:46:01:02:8a:98:c9:
    89:e5:d4:a3:2e:5b:bb:61:ff:33:8e:6f:f4:fe:7c:
    88:b3:9c:e2:ff:03:13:0b:4f:73:a1:32:f2:2e:e2:
    4c:9d:79:cb:28:38:13:81:f3:d8:12:d7:7b:80:12:
    5a:01:86:a5:d9:f5:1b:3a:99:2f:28:67:ca:8c:10:
    23:04:62:eb:8e:ac:05:3c:45:4b:93:51:98:56:a8:
    39:2d:41:ec:c0:02:24:c8:1d:e4:7c:ec:bb:cb:6b:
    79:69:91:11:7f:5e:18:e3:ab:e4:ae:21:b8:6b:e1:
    b4:46:52:5f:16:97:5e:b3:6b:62:b7:de:ce:7e:3d:
    e2:b1:28:1c:72:6d:cd:27:0a:22:66:55:cf:77:75:
    d8:74:e2:ea:e2:8d:af:d2:99:dc:1e:4b:6d:bb:c4:
    bd
P:   
    00:c3:1f:91:4d:12:c6:1f:a2:d3:66:b3:55:72:67:
    7d:2a:9a:2d:ac:1f:e7:65:37:64:ee:f6:e8:37:4d:
    0b:c6:59:98:d5:a7:1a:94:c1:db:1e:97:5b:3b:14:
    d9:32:58:da:d2:f3:9a:0c:a4:88:50:58:f0:12:d2:
    b4:0d:36:f5:a5:18:9f:62:ec:3f:6a:5b:90:4c:00:
    27:f2:53:1d:84:ba:6a:d1:8d:de:d7:83:09:7e:7c:
    fd:c7:de:e4:75:e0:2f:82:cb:e6:b6:cd:33:c1:a7:
    85:e3:f0:0a:fb:5b:9b:e9:cf:f2:6c:06:5e:fb:b9:
    3b:6e:15:76:00:f1:4e:ce:e3:3a:45:fc:1d:47:98:
    10:04:b6:a4:bc:ca:2a:1c:d8:82:17:fa:47:bd:cf:
    e1:9c:ae:ed:5d:25:22:0a:43:b8:3a:e1:ff:d1:2b:
    09:ef:b8:f8:65:03:22:eb:c6:c4:fc:08:d7:37:6c:
    ea:b4:8e:a0:c9:0a:5e:d1:70:84:d5:38:52:f1:59:
    7f:9d:01:a1:85:43:44:e6:31:68:00:b3:f1:1b:3c:
    2d:85:11:79:dd:37:27:db:4f:ed:d7:62:0d:ef:54:
    24:14:bf:f9:c6:fe:c0:5a:b8:5b:bf:a7:71:25:01:
    4c:f9:20:23:dd:88:0c:51:af:27:84:cc:c8:ed:dc:
    d4:43
Q:   
    00:88:d4:3b:9a:86:52:8b:cc:69:24:1b:7b:f1:e6:
    55:d7:e0:11:96:7b:35:5d:32:ae:ed:7d:05:89:9a:
    3c:e2:b3
G:   
    00:ab:8f:b5:93:bc:0d:10:15:bc:9e:b5:2e:8b:f3:
    67:ef:68:61:5d:3b:ca:78:44:4d:91:64:50:2f:c8:
    a8:ed:c6:60:50:fb:8a:f1:22:3e:5a:b3:b2:1d:6c:
    b4:aa:78:0e:c0:4b:95:da:11:c4:36:5a:7d:cd:be:
    24:50:39:93:40:ee:06:7e:0b:83:b6:b7:fe:80:94:
    86:2a:14:08:08:11:8d:ad:a2:09:4a:f3:9a:01:e0:
    8d:b8:3e:b6:4f:d2:d9:53:bc:e3:1e:d1:7d:75:3b:
    2c:ca:ae:20:b8:1e:ef:b7:dc:18:1a:d0:5d:30:46:
    dc:df:9a:0b:cf:5c:ad:77:41:ff:1d:a9:c6:a7:23:
    ae:cc:7a:7e:b5:03:9d:51:1b:fd:c0:0c:fb:0e:ed:
    b6:03:f2:92:af:eb:1e:70:c5:23:b3:ac:dc:cc:fd:
    86:b4:31:19:84:5f:23:4f:16:f4:ac:cb:0b:ba:77:
    46:c1:3a:0f:b1:69:36:ab:41:45:a7:41:8d:44:db:
    cf:07:dc:6b:80:fd:ca:8b:e6:fd:d7:b9:00:45:03:
    86:f9:85:b1:ba:b0:47:c3:8e:a3:40:42:7c:c8:38:
    95:30:a8:fb:50:e8:ee:98:e3:2b:03:39:9d:b7:07:
    a2:17:86:55:cd:ef:a3:02:56:a5:30:cf:0c:8e:15:
    73:35

### CODE REVIEW

**Comment by beldmit** [comment]
Please add a line "Fixes #9927" as the first line of your 1st comment.

**Comment by beldmit** [comment]
Could you also provide an output of the ```openssl pkey -text_pub -pubout``` command?

**Comment by jayaram-intel** [comment]
Hi @beldmit ,

Just now updated my comment and also provided output of the `openssl pkey -text_pub -pubout` command.

File: crypto/dsa/dsa_ameth.c [hunk 1]
```diff
@@ -361,6 +365,10 @@ static int do_dsa_print(BIO *bp, const DSA *x, int off, int ptype)
             <= 0)
             goto err;
     }
+    else{
```
Comment (beldmit) [inline crypto/dsa/dsa_ameth.c]: Nit. Merge the closing brace and newly provided one together and add the missing space between else and the opening brace.

Comment (jayaram-intel) [reply crypto/dsa/dsa_ameth.c]: Hi @beldmit ,
Please review the code with above mentioned changes.

Comment (beldmit) [reply crypto/dsa/dsa_ameth.c]: LGTM

**Comment by beldmit** [comment]
Merged. Thanks!

**Pull request closed without merging**